### ISSUE TITLE
Bug: SAM CLI 1.80 - authorizer 'None' is unsupported or not found

### ISSUE THREAD

**Issue by Barmil** [aws/aws-sam-cli#5038]
### Description:

I'm trying to test the new authorizer support in SAM CLI, which is super useful for black box tests.
I was trying to run an API gateway with a custom lambda authorizer for a specific route.
but I keep on getting "authorizer 'None' is unsupported or not found". which comes from [here](https://github.com/aws/aws-sam-cli/pull/5005/files#diff-f196b2a7b02ef167f8157d84352ebfecd0ee8290ffc72eb2c5222994df687922R111), since the function set `route.authorizer_name = None` before printing, it's a little bit hard to understand what's missing.
I was trying to set an Authorizer name just to clear it out of the way but in the end, I'm not sure what's wrong.

### Steps to reproduce:

Using CDK I'm creating:
```
const someAuthorizer = new RequestAuthorizer(scope, id, {
        authorizerName: id, // Added just in case
        handler: handler, // NodejsFunction
        identitySources: identitySources,
        resultsCacheTtl: resultsCacheTtl
    });
    
const someApiGatewayMethod = apiGatewayResource.addMethod(
  'GET',
  new LambdaIntegration(someLambda), {
    authorizationType: AuthorizationType.CUSTOM,
    authorizer: someAuthorizer
  })
```

### Observed result:

```
2023-04-18 16:20:39,181 | Linking authorizer skipped for route '<My_METHOD_FULL_ROUTE>', authorizer 'None' is unsupported or not found
```

### Expected result:

The custom authorizer will be attached and running using SAM CLI

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

  "version": "1.80.0",
  "system": {
    "python": "3.8.13",
    "os": "macOS-12.6.3-x86_64-i386-64bit"
  }

`Add --debug flag to command you are running`

**Comment by lucashuy** [comment]
Thanks for reporting this, from what I can see it doesn't look like the `Ref: <authorizer logical ID>` is being resolved correctly before being parsed. We'll have to look into this some more to see where the failure is happening. In the mean time, we can fix the log message not actually logging the name.

**Comment by mndeveci** [comment]
Released with [v1.82.0](https://github.com/aws/aws-sam-cli/releases/tag/v1.82.0)
